lib/lita/handlers/artifactory.rb

```
@@ -126,6 +126,11 @@ def promote(response)
           EOH
         end
 
+        # This event allows other lita utilities to act when projects are promoted.
+        robot.trigger(:artifactory_artifact_promoted, product: project, version: version)
```

We should also include the `source_channel` and `target_channel` if possible.

---

sure. easy enough.

---

Where would I find those though? The target channel is probably `PROMOTION_CHANNEL` but I'm not seeing a place for source channel?

---

Ahh you are correct, `source_channel` isn't returned in the response from:
https://github.com/chef-cookbooks/oc-artifactory/blob/master/files/default/plugins/Promotions.groovy

I suppose a simple map in that class could handle the source => target mapping....we know the promotion shape:

```ruby
def source_channel(target_channel)
  case target_channel
  when 'current'; 'unstable'
  when 'stable'; 'current'
  else raise "Source channel is not known for target channel: #{target_channel}"
  end
end```

---

Would we really want to raise an error when sending an event? We could just specify "unknown" or something...

---

Also, isn't `@julia artifactory promote` only support promotion from current to stable?

---

@tduffield Yep. Go with what you think is the best fix....was just sharing `def source_channel` as an example.